client/my-sites/email/form/mailboxes/components/field.tsx

```diff
@@ -91,11 +98,11 @@ const MailboxField = ( {
 		}
 		field.value = value;
 
-		// Validate the field on the fly if there was already an error
-		if ( field.error ) {
+		// Validate the field on the fly if there was already an error, or the field was already touched
+		if ( field.error || field.isTouched ) {
```

Comment:
Do we really need to look at `field.isTouched` here? If we received a `change` event, it seems implied that the field has been touched, right..?

Reply:
Or I guess the idea is that if I focus a field, fill in a value, blur the field, then refocus the field, I should see the result of all validators immediately. Is that correct?

Reply:
> Do we really need to look at field.isTouched here? If we received a change event, it seems implied that the field has been touched, right..?

Mostly yes. However, you sometimes want to differentiate from when the `onChange` has previously occurred while you are in the event, and that is difficult to do except you store that in the field state. 

Secondly there are interactions that also want to execute logic based on if a field has been `touched` (before). This also mean we need to store that state so that we can access it later.

Reply:
> Or I guess the idea is that if I focus a field, fill in a value, blur the field, then refocus the field, I should see the result of all validators immediately. Is that correct?

The validation should happen after the first time the field is blurred out ( i.e. a refocus is not need, so that the user attempts to make correction before they move on to the next field ), then subsequently almost immediately